### PR TITLE
[FragmentStrictMode] Detect Fragment reuse

### PR DESCRIPTION
**Pull request by simonschiller**
## Proposed Changes

  - Detect reuse of `Fragment` instances

## Testing

Test: See `FragmentStrictModeTest#detectFragmentReuse` and `FragmentStrictModeTest#detectFragmentReuseInFlightTransaction`

## Issues Fixed

Fixes: 153738653

### CODE REVIEW

File: fragment/fragment/src/main/java/androidx/fragment/app/Fragment.java [hunk 1]
```diff
@@ -283,6 +283,10 @@ public void run() {
     // track it separately.
     boolean mIsCreated;
 
+    // True if the fragment was already added to a FragmentManager, but has since been removed
+    // again.
+    boolean mRemovedFromFragmentManager;
```
Comment (jbw0033) [inline fragment/fragment/src/main/java/androidx/fragment/app/Fragment.java]: This name is long, I think `mRemoved` would be enough.

**Review by jbw0033 — CHANGES_REQUESTED** [review]

**Comment by dlam** [comment]
Build should be passing again now - if you rebase this checks should pass.

**Comment by simonschiller** [comment]
Rebased and the checks here seem fine, but I got an email notification telling me that they didn't pass. Is this branch green on your internal CI?

**Comment by dlam** [comment]
I think the lint check has some issues with api files right now on push events - feel free to ignore since checks here are passing.

**Review by jbw0033 — APPROVED** [review]

**Pull request closed without merging**